sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcUtils.scala

```diff
@@ -372,12 +386,14 @@ object OrcUtils extends Logging {
       canPruneCols: Boolean,
       dataSchema: StructType,
       resultSchema: StructType,
+      orcCatalystSchema: StructType,
       partitionSchema: StructType,
       conf: Configuration): String = {
     val resultSchemaString = if (canPruneCols) {
-      OrcUtils.orcTypeDescriptionString(resultSchema)
+      OrcUtils.orcTypeDescriptionString(resultSchema, Some(orcCatalystSchema))
```

Comment:
It's quite hacky to put schema evolution code in a method named `orcTypeDescriptionString`. I'd prefer more explicit code similar to `ParquetReadSupport.clipParquetSchema`